cmd/crank/beta/trace/trace.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package trace contains the trace command.
+package trace
+
+import (
+	"context"
+
+	"github.com/alecthomas/kong"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/restmapper"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/crossplane/crossplane/cmd/crank/beta/trace/internal/printer"
+	"github.com/crossplane/crossplane/cmd/crank/beta/trace/internal/resource"
+)
+
+const (
+	errGetResource        = "cannot get requested resource"
+	errCliOutput          = "cannot print output"
+	errKubeConfig         = "failed to get kubeconfig"
+	errInitKubeClient     = "cannot init kubeclient"
+	errGetDiscoveryClient = "cannot get discovery client"
+	errGetMapping         = "cannot get mapping for resource"
+	errInitPrinter        = "cannot init new printer"
+)
+
+// Cmd builds the trace tree for a Crossplane resource.
+type Cmd struct {
+	Resource string `arg:"" help:"Kind of the of the Crossplane resource, accepts the 'TYPE[.VERSION][.GROUP]' format."`
+	Name     string `arg:"" help:"Name of the Crossplane resource."`
+
+	// TODO(phisco): add support for all the usual kubectl flags; configFlags := genericclioptions.NewConfigFlags(true).AddFlags(...)
+	// TODO(phisco): move to namespace defaulting to "" and use the current context's namespace
+	Namespace             string `short:"n" name:"namespace" help:"Namespace of the resource." default:"default"`
+	Output                string `short:"o" name:"output" help:"Output format. One of: default, wide, json, dot." enum:"default,wide,json,dot" default:"default"`
+	ShowConnectionSecrets bool   `short:"s" name:"show-connection-secrets" help:"Show connection secrets in the output."`
+}
+
+// Help returns help message for the trace command.
+func (c *Cmd) Help() string {
+	return `
+This command trace a Crossplane resource to get a detailed output of its
```

Comment:
```suggestion
This command trace a Crossplane resource (Claim, Composite, or Managed Resource) to get a detailed output of 
```

It would be helpful to be explicit about what we mean by "Crossplane resource". Not sure tracing an MR would make sense at all but AFAIU command works with them as well.

Reply:
done: https://github.com/crossplane/crossplane/pull/4849/commits/f8998ceee913e53c1c4585e448f5d0936c44fb38